pkg/server/datastore/sqlstore/sqlstore.go

```diff
@@ -427,7 +444,15 @@ func (ds *Plugin) DeleteRegistrationEntry(ctx context.Context,
 ) (registrationEntry *common.RegistrationEntry, err error) {
 	if err = ds.withWriteTx(ctx, func(tx *gorm.DB) (err error) {
 		registrationEntry, err = deleteRegistrationEntry(tx, entryID)
-		return err
+		if err != nil {
+			return err
+		}
+
+		if fflag.IsSet(fflag.FlagEventsBasedCache) {
+			return createRegistrationEntryEvent(tx, entryID)
```

Comment:
non-blocking nit: i think the code gets a little cleaner if createRegistrationEntryEvent checks the flag (and returns nil immediately if not set 🤷 )

Reply:
Fixed